src/test/java/com/parasoft/demoapp/graphql/ItemGraphQLDataFetcherTest.java

```diff
@@ -207,4 +243,142 @@ public void test_updateItemInStockByItemId_noAuthentication() throws IOException
+        GraphQLResponse response = graphQLTestTemplate
+                .withBasicAuth(USERNAME_PURCHASER, PASSWORD)
+                .perform(ADD_NEW_ITEM_GRAPHQL_RESOURCE, variables);
+        assertThat(response).isNotNull();
+        assertThat(response.isOk()).isTrue();
+        response.assertThatErrorsField().isNotNull()
+                .asListOf(GraphQLTestError.class)
+                .hasOnlyOneElementSatisfying(error -> {
+                    assertThat(error.getMessage()).isEqualTo(AssetMessages.ITEM_NAME_EXISTS_ALREADY);
+                    assertThat(error.getExtensions().get("statusCode")).isEqualTo(HttpStatus.BAD_REQUEST.value());
+                })
+                .and()
+                .assertThatField(ADD_NEW_ITEM_DATA_JSON_PATH).isNull();
+    }
+
+    @Test
+    public void test_addNewItem_categoryNotFoundException() throws IOException {
+        ObjectNode variables = objectMapper.createObjectNode().putPOJO("itemsDTO",
+                new ItemsDTO("Tent", "description", 50L, 10, null, LOCATION_1));
+        GraphQLResponse response = graphQLTestTemplate
+                .withBasicAuth(USERNAME_PURCHASER, PASSWORD)
+                .perform(ADD_NEW_ITEM_GRAPHQL_RESOURCE, variables);
+        assertThat(response).isNotNull();
+        assertThat(response.isOk()).isTrue();
+        response.assertThatErrorsField().isNotNull()
+                .asListOf(GraphQLTestError.class)
+                .hasOnlyOneElementSatisfying(error -> {
+                    assertThat(error.getMessage()).isEqualTo(MessageFormat.format(AssetMessages.CATEGORY_ID_NOT_FOUND, 50L));
+                    assertThat(error.getExtensions().get("statusCode")).isEqualTo(HttpStatus.NOT_FOUND.value());
+                })
+                .and()
+                .assertThatField(ADD_NEW_ITEM_DATA_JSON_PATH).isNull();
+    }
+
+    @Test
+    public void test_addNewItem_invalidOrNullItemNameValue() throws IOException {
+        ObjectNode variables = objectMapper.createObjectNode().putPOJO("itemsDTO",
+                new ItemsDTO("", "description", 1L, 10, null, LOCATION_1));
+        GraphQLResponse response = graphQLTestTemplate
+                .withBasicAuth(USERNAME_PURCHASER, PASSWORD)
+                .perform(ADD_NEW_ITEM_GRAPHQL_RESOURCE, variables);
+        assertThat(response).isNotNull();
+        assertThat(response.isOk()).isTrue();
+        response.assertThatErrorsField().isNotNull()
+                .asListOf(GraphQLTestError.class)
+                .hasOnlyOneElementSatisfying(error -> {
+                    assertThat(error.getMessage()).isEqualTo(AssetMessages.ITEM_NAME_CANNOT_BE_BLANK);
+                    assertThat(error.getExtensions().get("statusCode")).isEqualTo(HttpStatus.BAD_REQUEST.value());
+                })
+                .and()
+                .assertThatField(ADD_NEW_ITEM_DATA_JSON_PATH).isNull();
+    }
+
+    @Test
+    public void test_addNewItem_regionNotExistOnCurrentIndustry() throws IOException {
```

Comment:
Add one more case to pass in invalid region name which is not a valid RegionType. For example: JAPAN123

Reply:
It is specified here that only RegionType enumeration types can be passed. I can add a case where the region name is empty，and the results of the two are the same